### PR TITLE
Faster lookup

### PR DESCRIPTION
**Pull request by arjo129**
This PR depends on:
* https://github.com/ignitionrobotics/ign-math/pull/406

It supersedes #188 and only focuses on the lookup thus making the reviewers life easier.

It implements a fast lookup for grid data like the data provided by MBARI. It also adds an integration test for the science data. Currently we are only checking data that does not vary with time. I have removed all interpolation code from this PR to keep it clean. I will be adding it back in the next PR. On my Laptop's Ryzen 9 5900HX this code takes between 3-15 microseconds per lookup.

### CODE REVIEW

File: lrauv_ignition_plugins/src/ScienceSensorsSystem.cc [hunk 1]
```diff
@@ -1498,231 +670,82 @@ void ScienceSensorsSystem::PostUpdate(const ignition::gazebo::UpdateInfo &_info,
...
+    //TODO(arjo): Replace with interpolation code.
+    //For now just report some measurement.
+    IGN_PROFILE("ScienceSensorsSystem::Interpolation");
+    if (interpolators.size() == 0) return;
+    if (!interpolators[0].has_value()) return;
+
+    if (auto casted = std::dynamic_pointer_cast<SalinitySensor>(sensor))
     {
-      break;
+      float sal =
```
Comment (hidmic) [inline lrauv_ignition_plugins/src/ScienceSensorsSystem.cc:718]: @arjo129 nit: can be made `const`, same for most below.

**Review by hidmic — COMMENTED** [review]
First pass. Left some comments, most are minor. I understand this is halfway a refactor, so I'm good with deferring code cleanup.

File: lrauv_ignition_plugins/test/test_sensor.cc [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+  // Check that vehicles don't exist
+  fixture->Server()->RunOnce();
+  EXPECT_EQ(ignition::gazebo::kNullEntity, vehicle1);
+  EXPECT_EQ(1, iterations);
+  EXPECT_EQ(0, poses1.size());
+  EXPECT_EQ(0, latLon1.size());
+
+  // Spawn first vehicle
+  ignition::transport::Node node;
+  node.Subscribe("/model/vehicle1/chlorophyll", &ChlorophyllCb);
```
[the remark this answers is not in the document]
Comment (arjo129) [reply lrauv_ignition_plugins/test/test_sensor.cc:90]: Actually, I eventually want to eliminate the zoo off worlds we have and focus just on `empty_environment.sdf`. Ideally we would write spawning infra to make it easy. For now I have not done that yet.

[the remark this answers is not in the document]
Comment (hidmic) [reply lrauv_ignition_plugins/test/test_sensor.cc:90]: @arjo129 I agree with not having lots of worlds that are only slightly different. At the same time, I'm not sure if we can (should?) accommodate every test in the same world e.g. for some tests we may have to ensure there are no external forces acting on the vehicle, while others require them. Crafting the world in runtime also adds unrelated failure modes to a test, like in this case, where 90%+ of the assertions are accessory to what is being tested.

In my mind, having a generic base SDF and extending it with test-specific SDF snippets would be ideal. Unclear if that's possible though.

**Comment by hidmic** [comment]
I'll re-review when we merge #191 in.

**Pull request closed without merging**